packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx

```diff
@@ -82,13 +80,16 @@ const Zone = ({
     setEditModalData({ isOpen: false });
   };
 
+  /** @param {keyof typeof Widgets} widgetTypeName */
   const onSelectFieldType = (widgetTypeName) => {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-assignment
-    setNewFieldData({ widgetTypeName, fields });
+    /** `widgetTypeName` might have less keys than `Widgets`, but we lost track 
+    of the types because the `widgetsArray` is not typed and is also filtered into 
+    `widgetsArrayWithCondUid`. Although it's safe to use it to index the `Widgets` 
```

Comment:
Minor grammar & spelling fixes in italics:
"_Although,_ it's safe to use it to index the `Widgets` as _long_ as..."

Reply:
Corrected ✅